card/lib/card/location.rb

```diff
@@ -0,0 +1,45 @@
+# -*- encoding : utf-8 -*-
+# helper for urls, links, redirects, and other location related things.
+#  note: i'm sure this isn't the optimal name..
+
+module Card::Location
+
+ # -----------( urls and redirects from application.rb) ----------------
+
+  # TESTME
+  def page_path title, opts={}
+    
+    format = opts[:format] ? ".#{opts.delete(:format)}"  : ''
+    query  = opts.present? ? "?#{opts.to_param}"         : ''
+    card_path "#{title.to_name.url_key}#{format}#{query}"
+  end
+
+  def card_path rel #should be in smartname?
+    rel_path = Card===rel ? rel.cardname.url_key : rel.to_s
+    if rel_path =~ /^\//
+      rel_path
+    else
+      "#{ Cardio.config.relative_url_root }/#{ rel_path }"
+    end
+  end
+
+  def card_url rel #should be in smartname?
+    if rel =~ /^https?\:/
+      rel
+    else
+      "#{ Card::Env[:protocol] }#{ Card::Env[:host] }#{ card_path rel }"
+    end
+  end
+
+
+  # Links ----------------------------------------------------------------------
+
+  def link_to_page( text, title=nil, options={})
+    title ||= text
+    url_options = {}
+    [:type, :view].each { |k| url_options[k] = options.delete(k) if options[k] }
+    url = page_path( title, url_options )
+    link_to text, url, options
+  end
+
```

Comment:
You understand that some of the stuff in Wagn::Location moved to Decko::Location, right?  Essentially the session stuff because that is bound to the controller environment.  I just explicitly require 'decko/location' in mods that use sessions.  Here we are only talking about the Card::Location methods.  We should discuss if you think we need this to be done differently before we can merge.  I think it is ok for some modules to have bindings to decko.  We would have to adapt those parts for a different web-context, but it probably isn't that hard to generalize this important touch point.  Could be as simple as being able to configure a replacement for Decko::Location via Cardio.